doc/user-guide/data_model.rst

```diff
@@ -0,0 +1,251 @@
+excellent C array library `NumPy <https://numpy.org/>`_.  You could
+more create a points array with:
+
+.. jupyter-execute::
+
+   >>> import numpy as np
+   >>> np_points = np.array([[0, 0, 0],
+   ...                       [1, 0, 0],
+   ...                       [0.5, 0.667, 0]])
+   >>> np_points
+
+We use a :class:`numpy.ndarray` here so that PyVista directly "point"
+the underlying C array to VTK.  VTK already has APIs to directly read
+in the C arrays from ``numpy``, and since VTK is written in C++,
+everything from Python that is transferred over to VTK needs to be in a
+format that VTK can process.
+
+Should you wish to use VTK objects within PyVista, you can still do
+this.  In fact, using :func:`pyvista.wrap`, you can even get a numpy-like
+representation of the data.  For example:
+
+.. jupyter-execute::
+
+   >>> import pyvista
+   >>> wrapped = pyvista.wrap(vtk_array)
+   >>> wrapped
+
+Note that when wrapping the underlying VTK array, we actually perform
+a shallow copy of the data.  In other words, we pass the pointer from
+the underlying C array to the numpy :class:`numpy.ndarray`, meaning
+that the two arrays are now efficiently linked.  This means that we
+can change the array using numpy array indexing and have it modified
+on the "VTK side".
+
+.. jupyter-execute::
+
+   >>> wrapped[0, 0] = 10
+   >>> vtk_array.GetValue(0)
+
+Or we can change the value from the VTK array and see it reflected in
+the numpy wrapped array.  Let's change the value back:
+
+.. jupyter-execute::
+
+   >>> vtk_array.SetValue(0, 0)
+   >>> wrapped[0, 0]
+
+
+Using a Python List
+~~~~~~~~~~~~~~~~~~~
+PyVista supports the use of Python lists, and you could define a your
+points using a nested list of lists via:
+
+.. jupyter-execute::
+
+   >>> points = [[0, 0, 0],
+   ...           [1, 0, 0],
+   ...           [0.5, 0.667, 0]]
+
+When used in the context of :class:`pyvista.PolyData` to create the
+mesh, this list will automatically be wrapped using numpy and then
+passed to VTK.  This avoids any looping overhead and while still
+allowing you to use native python classes.
+
+Finally, let's show how we can use these three objects in the context
+of a PyVista geometry class.  Here, we create a simple point mesh
+containing just the three points:
+
+.. jupyter-execute::
+   
+   >>> from_vtk = pyvista.PolyData(vtk_array)
+   >>> from_np = pyvista.PolyData(np_points)
+   >>> from_list = pyvista.PolyData(points)
+
+These point meshes all contain three points and are effecively
+identical.  Let's show this by accessing the underlying points array
+from the mesh, which is represented as a :class:`pyvista_ndarray`
+
+.. jupyter-execute::
+
+   >>> from_vtk.points
+
+And show that these are all identical
+
+.. jupyter-execute::
+
+   >>> assert np.allclose(from_vtk.points, from_np.points)
+   >>> assert np.allclose(from_vtk.points, from_list.points)
+   >>> assert np.allclose(from_np.points, from_list.points)
```

Comment:
There's no arithmetic going on (hopefully), so this should be true with `np.array_equal`, which I'd slightly prefer here.

Reply:
Added